File: creational/singleton/singleton.py

# Singleton class
class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Singleton, cls).__new__(cls)
        return cls._instance

    def do_something(self):
        print("Singleton instance is doing something.")

if __name__ == "__main__":
    instance1 = Singleton()
    instance1.do_something()

    instance2 = Singleton()
    print(instance1 is instance2)  # True

File: creational/singleton/Singleton.cs

using System;

// Singleton class
public class Singleton
{
    private static Singleton _instance;

    // Private constructor to prevent instantiation
    private Singleton() { }

    public static Singleton GetInstance()
    {
        if (_instance == null)
        {
            _instance = new Singleton();
        }
        return _instance;
    }

    public void DoSomething()
    {
        Console.WriteLine("Singleton instance is doing something.");
    }
}

// Client code
class Program
{
    static void Main()
    {
        Singleton instance1 = Singleton.GetInstance();
        instance1.DoSomething();

        Singleton instance2 = Singleton.GetInstance();
        Console.WriteLine(object.ReferenceEquals(instance1, instance2)); // True
    }
}
